feat(demo): Demo for drawer mode with display mode switcher (#601)
patternfly/chatbot: packages/module/patternfly-docs/content/extensions/chatbot/examples/demos/Chatbot.md

```

> ### Display mode ChatBot

Reviewer: ```suggestion
### Display mode switcher
```

This demo showcases how the ChatBot can be rendered in different display modes to suit various application layouts. It demonstrates how to dynamically change the page structure in response to the user's selection. This demo includes:

> 1. The ability to switch between Overlay, Docked, and Fullscreen modes using the [`<ChatbotHeaderOptionsDropdown>`](/patternfly-ai/chatbot/ui#header-options) in the header.

Reviewer: ```suggestion
1. The ability to switch between overlay, drawer, and fullscreen modes using the [`<ChatbotHeaderOptionsDropdown>`](/patternfly-ai/chatbot/ui#header-options) in the header.
```

> 2. A conditional page layout that renders the ChatBot:

Reviewer: ```suggestion
2. A conditional page layout that renders the ChatBot for each display mode option:
```

>     - As a floating window on top of the page content (Overlay mode).
>     - Inside a PatternFly `<Drawer>` as a side panel for a docked experience.
>     - As a top-level component that covers the entire screen (Fullscreen mode).

Reviewer: ```suggestion
    - **Overlay:** As a floating window on top of the page content.
    - **Drawer:** Inside an inline PatternFly `<Drawer>` as a side panel.
    - **Fullscreen:** As a top-level component that covers the entire screen for an embedded experience.
```
Might rephrase like this for easier scanning

> 3. Logic to show or hide the `<ChatbotToggle>` button, which is only present in the default Overlay mode.

Reviewer: ```suggestion
3. Logic to show or hide the `<ChatbotToggle>` button, which is only present in the default overlay mode.
```
